### PR TITLE
Keep tmpfiles/legacy.conf even if SysVInit support is dropped

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: tmpfiles.d/legacy.conf.in [hunk 1]
```diff
@@ -7,24 +7,25 @@
...
 # See tmpfiles.d(5) for details
 
-# These files are considered legacy and are unnecessary on legacy-free
-# systems.
+# The functionality provided by these files and directories has been replaced
+# by newer interfaces. The configurationhere is provided to maintain backwards
+# compatibility. Their use is discouraged on legacy-free systems.
 
 d /run/lock 0755 root root -
 L /var/lock - - - - ../run/lock
```
Comment (poettering) [inline tmpfiles.d/legacy.conf.in:15]: hmm, what stuff is still using these dirs?

Comment (keszybz) [reply tmpfiles.d/legacy.conf.in:15]: Apparently lots. Alsa, iscsi, lvm on my Fedora system. But the issue is that we can't really know, there can always be things written 20 years ago that use `/var/lock`. We *could* break those things, but it doesn't seem worth the trouble. There is no particular problem with this directories and the cost of keeping them is very little. On most systems they'll just sit unused.

File: tmpfiles.d/legacy.conf.in [hunk 2]
```diff
@@ -5,26 +5,27 @@
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
 
-# See tmpfiles.d(5) for details
+# See tmpfiles.d(5) for details.
 
-# These files are considered legacy and are unnecessary on legacy-free
-# systems.
+# The functionality provided by these files and directories has been replaced
+# by newer interfaces. The configurationhere is provided to maintain backwards
```
Comment (YHNdnzj) [inline tmpfiles.d/legacy.conf.in]:
```suggestion
# by newer interfaces. The configuration here is provided to maintain backwards
```

File: tmpfiles.d/legacy.conf.in [hunk 3]
```diff
@@ -5,26 +5,27 @@
...
 d /run/lock/subsys 0755 root root -
+{% endif %}
 
 # /forcefsck, /fastboot and /forcequotacheck are deprecated in favor of the
 # kernel command line options 'fsck.mode=force', 'fsck.mode=skip' and
 # 'quotacheck.mode=force'
-
 r! /forcefsck
 r! /fastboot
 r! /forcequotacheck
```
Comment (YHNdnzj) [inline tmpfiles.d/legacy.conf.in:30]: Hrm, these checks are conditionalized out on HAVE_SYSV_COMPAT in relevant C code too. So move this into if block?

File: tmpfiles.d/systemd-nspawn.conf [hunk 4]
```diff
@@ -5,7 +5,7 @@
 #  the Free Software Foundation; either version 2.1 of the License, or
 #  (at your option) any later version.
 
-# See tmpfiles.d(5) for details
+# See tmpfiles.d(5) for details.
```
Comment (YHNdnzj) [inline tmpfiles.d/systemd-nspawn.conf:8]: BTW, this file should probably be renamed to systemd-machine, given we now have vmspawn that makes use of these dirs.

**Comment by YHNdnzj** [comment]
So I think this makes sense to be merged in v257. Effectively only Arch is affected, and is a nice way forward if other distros want to disable sysv compat already in v257.

**Comment by eworm-de** [comment]
> So I think this makes sense to be merged in v257. Effectively only Arch is affected, and is a nice way forward if other distros want to disable sysv compat already in v257.

Yes, please. 👍

File: tmpfiles.d/legacy.conf.in (hunk 3 above)
Comment (keszybz) [reply tmpfiles.d/legacy.conf.in:30]: Yeah, I think we should. Updated like that.

File: tmpfiles.d/systemd-nspawn.conf (hunk 4 above)
Comment (keszybz) [reply tmpfiles.d/systemd-nspawn.conf:8]: Renaming of files is problematic… People can have masks or oerrides and those break when we change the name. So I think we should just keep the name, even if it doesn't fit exactly, unless there's a strong reason to rename.

Comment (YHNdnzj) [reply tmpfiles.d/systemd-nspawn.conf:8]: Indeed. Please ignore then.

**Review by YHNdnzj — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Comment by keszybz** [comment]
ci (centos, 10, 0, -Og, yes) is the old TEST-55-OOMD thing.

**Comment by keszybz** [comment]
OK, let's merge this without waiting for ubuntuautopkgtests. They are built with HAVE_SYSV_COMPAT, so it'd be very strange if they were affected by this PR at all.

**Pull request merged**